core/src/main/java/io/grpc/internal/ManagedChannelImpl.java

```diff
@@ -582,6 +592,53 @@ private void maybeTerminateChannel() {
     }
   }
 
+  @Override
+  public ConnectivityState getState(boolean requestConnection) {
+    ConnectivityState savedChannelState = channelState;
+    if (requestConnection && savedChannelState == IDLE) {
+      channelExecutor.executeLater(new Runnable() {
+        @Override
+        public void run() {
+          exitIdleMode();
+        }
+      }).drain();
+    }
+    return savedChannelState;
+  }
+
+  private static final class StateChangeCallBack implements Runnable {
+    final ConnectivityState source;
+    final Runnable callback;
+    boolean done;
```

Comment:
Even if the field were marked as `private`, it would still be accessible from `ManagedChannelImpl`. To make it worse, compiler will create an accessor for this private field to allow the outer class to access it, leading to larger code size.